packages/blocks/src/api/registration.js

```diff
@@ -160,6 +160,11 @@ export const serverSideBlockDefinitions = {};
 // eslint-disable-next-line camelcase
 export function unstable__bootstrapServerSideBlockDefinitions( definitions ) {
 	for ( const blockName of Object.keys( definitions ) ) {
+		// Don't overwrite if already set. It covers the case when metadata
+		// was initialized from the server.
+		if ( serverSideBlockDefinitions[ blockName ] ) {
```

Comment:
What's a little fragile here is depending on the load order remaining the same, but I think the E2E should catch the behavior change, if for some reason in the future the client file was read first.

Reply:
Right, I didn't think about it before. Looking at how it's processed in WordPress core, it should be reliable:

https://github.com/WordPress/wordpress-develop/blob/d271a206dbc87421b221edb2f7acff6ea523fb5d/src/wp-admin/edit-form-blocks.php#L117-L120

It's glued to `wp-block` script handle, that `wp-block-library` script defines as a dependency.